src/theory/strings/sequences_rewriter.cpp

```diff
@@ -1662,132 +1766,125 @@ Node SequencesRewriter::rewriteMembership(TNode node)
-  }
-  else if (x != node[0] || r != node[1])
-  {
-    retNode = NodeManager::currentNM()->mkNode(kind::STRING_IN_REGEXP, x, r);
+    Node retNode = nm->mkNode(STRING_IN_REGEXP, x, r[0]).negate();
+    return returnRewrite(node, retNode, Rewrite::RE_IN_COMPLEMENT);
   }
 
   // do simple consumes
-  if (retNode == node)
+  Node retNode = node;
+  if (r.getKind() == kind::REGEXP_STAR)
   {
-    if (r.getKind() == kind::REGEXP_STAR)
+    for (unsigned dir = 0; dir <= 1; dir++)
     {
-      for (unsigned dir = 0; dir <= 1; dir++)
+      std::vector<Node> mchildren;
+      utils::getConcat(x, mchildren);
+      bool success = true;
+      while (success)
       {
-        std::vector<Node> mchildren;
-        utils::getConcat(x, mchildren);
-        bool success = true;
-        while (success)
+        success = false;
+        std::vector<Node> children;
+        utils::getConcat(r[0], children);
+        Node scn = simpleRegexpConsume(mchildren, children, dir);
+        if (!scn.isNull())
+        {
+          Trace("regexp-ext-rewrite")
+              << "Regexp star : const conflict : " << node << std::endl;
+          return returnRewrite(node, scn, Rewrite::RE_CONSUME_S_CCONF);
+        }
+        else if (children.empty())
         {
-          success = false;
-          std::vector<Node> children;
-          utils::getConcat(r[0], children);
-          Node scn = simpleRegexpConsume(mchildren, children, dir);
-          if (!scn.isNull())
+          // fully consumed one copy of the STAR
+          if (mchildren.empty())
           {
             Trace("regexp-ext-rewrite")
-                << "Regexp star : const conflict : " << node << std::endl;
-            return scn;
+                << "Regexp star : full consume : " << node << std::endl;
+            Node ret = NodeManager::currentNM()->mkConst(true);
+            return returnRewrite(node, ret, Rewrite::RE_CONSUME_S_FULL);
           }
-          else if (children.empty())
+          else
           {
-            // fully consumed one copy of the STAR
-            if (mchildren.empty())
-            {
-              Trace("regexp-ext-rewrite")
-                  << "Regexp star : full consume : " << node << std::endl;
-              return NodeManager::currentNM()->mkConst(true);
-            }
-            else
-            {
-              retNode = nm->mkNode(STRING_IN_REGEXP,
-                                   utils::mkConcat(mchildren, stype),
-                                   r);
-              success = true;
-            }
+            retNode = nm->mkNode(
+                STRING_IN_REGEXP, utils::mkConcat(mchildren, stype), r);
+            success = true;
           }
         }
-        if (retNode != node)
-        {
-          Trace("regexp-ext-rewrite") << "Regexp star : rewrite " << node
-                                      << " -> " << retNode << std::endl;
-          break;
-        }
+      }
+      if (retNode != node)
```

Comment:
If I am not mistaken, we should add a rewrite enum for this case and return instead of break.

Reply:
Good catch, fixed.